include/highfive/H5DataSpace.hpp

```diff
@@ -150,14 +149,15 @@ class DataSpace : public Object {
 #endif
 
   protected:
-    explicit DataSpace();
+    DataSpace() = default;
 
     friend class Attribute;
     friend class File;
     friend class DataSet;
 };
 }
 
+// We include bits right away since DataSpace can be constructed stand-alone
```

Comment:
so why do we also include bits of `H5Object` ? I would think that class is not contructed stand-alone? Unless I misunderstand the meeting of _stand-alone_ here.

Reply:
Ok I didn’t apply the simplification rules to classes whose implementation is self contained, like H5object, and therefore not prone to be problematic. Otherwise we would have to track everywhere where h5object.hpp is used and include additionally h5object_misc. I’ll add a note

Reply:
True, stand-alone was not the best wording. It's user constructible, and that's a reason strong enough.
Btw, H5Object also brings implementation since it's a fundamental block, I think it's the only exception.